Initial BLIP3 proposal: Peer Review

atfornes opened the pull request:
first draft

Co-authored-by: Elena Pérez Tirador <ElenaPT@users.noreply.github.com>
Co-authored-by: Kaan Uzdogan<kuzdogan@users.noreply.github.com>
Co-authored-by: Paulo Colombo <PJColombo@users.noreply.github.com>

Commit:
Initial BLIP3 proposal: Peer Review

first draft

Co-authored-by: <ElenaPT@users.noreply.github.com>
Co-authored-by: <kuzdogan@users.noreply.github.com>
Co-authored-by: <PJColombo@users.noreply.github.com>

kiview commented on blips/blip-3-peerreview.md:
```diff
@@ -0,0 +1,53 @@
+
+| Field | data type | Format/s | Mandatory |  Comments|
+| -------- | ---- | ---- | -------- |----|
+|     id | string  | IPFS Hash??   | Yes     | The id should be unique. How to make sure reviews are unique? See [Zooko's triangle](https://en.wikipedia.org/wiki/Zooko%27s_triangle) for a relevant discussion on the compromises between decentralization, human usability and unique identifiers   |
```
Does this need to be a universally unique ID (so UUID), or just unique in the context of the peer review record domain? We also have the transaction hash as a kind of unique identifier. Or is the idea to keep in constant in case of updates?

atfornes replied:
Thanks @kiview, this is an interesting discussion to have. From my point of view, there should not be duplicate reviews in the system, meaning that a given review, by a given author for a given manuscript should be only included once in the system. Ideally, the identifier should be unique for that purpose. The hash of the first transaction referring to that id could be indeed a great candidate. If we want to allow updates, an IPFS address of other hashes are not ideal identifiers.

This also opens a new discussion, how to deal with duplicate reviews in the system?

kiview replied:
I think we can conceptually differentiate between two things:

1. Having 1 logical instance of a peer-review. 
2. Having multiple physical instances of a peer-review record saved on the ledger. 

Multiple records on the ledger can very well reflect (auditable) updates to the record, so this is of course fine. However, how can we ensure that a bad actor isn't publishing multiple instances of the same logical peer-review, with each getting a new unique id? Do we need a locality sensitive hash for this? 

We are in touch with the folks from https://iscc.codes/ for another BLIP, maybe this approach can help here as well (very similar use case actually).

atfornes replied:
I don't think hashes can help us there. Maybe we need some ways to bring trust to the provenance of the peer review record. There is already a mechanism to do it with vouchers. We could promote trustworthy actors to vouche for reviews, such as Journals, maybe authors, and maybe other review registries (Orcid, Publons, F1000, Crossref, etc).

As an advance strategy, we could use something like a Token Curated Registry to vouch for identities, Only trusted identities will have strong weight when vouching, and they could loose trust if they misbehave.

JamesLawton replied:
@atfornes Agreed - I don't believe hashes can solve the problem of bad actors in this case. The preliminary voucher (Endorser :P) system we developed is quite simplistic, needing only a single other person to endorse the review in order for it to become valid. I like the system in general because it takes advantage of the decentralization of the blockchain, but also social networking effects to hopefully make the app more attractive and interactive. The caveat is that it needs a more robust implementation in its current state (weighting mechanism, 'trusted' endorser, etc).

kiview commented on blips/blip-3-peerreview.md:
```diff
@@ -0,0 +1,53 @@
+|     vouchersMap | mapping(address => bool)     |     |  Yes    |
+
+
+- [ ] **TODO** Decide which field could be private. For instance, the review content could be publicly shared or not depending on the reviewer's preferences.
```
Private in the sense of visibility in the contract code? Data itself will still be more or less public. AFAIK we can only tackle real privacy here by encryption or sharing data off-chain.

atfornes replied:
Private in the sense of openly shared in a blockchain or not shared in a blockchain (but maybe yes with specific apps, service providers, etc)

I am thinking mainly about two content that could be either shared publicly or only with some parties:

- email addresses
- Peer review report content

But I am sure there are many other cases.

Taking inspiration by what Orcid does, they only share a URL of the review, then that URL can openly share the authors identity, the content of the review or neither. see https://support.orcid.org/hc/en-us/articles/360006971333-Peer-Review

kiview replied:
I agree with this, just want to be sure we are on the same page when we say, private means it has to be stored off-chain. This is slightly touching the SSI domain.

atfornes commented:
It would be great to create a branch named `blip3` where we can update this Blip proposal. Could someone with write privileges in this repo create such branch?

kiview commented:
@atfornes This is a good point and TBH not something I completely thought through. I can create a branch for this, but then only the repo maintainers would have write permissions to it.

Or you can give the maintainers write permissions to this branch in your fork:
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Ultimately both approaches are not perfect and now I wonder, how other projects handle this more elegantly. I think it all boils down to maximum author + maintainers having write access. At least in the PR review, everyone can include changes as suggestions in code comments.

atfornes commented:
I cannot allow edits for maintainers at this pull request. please modify the [hackmd file](https://hackmd.io/QtoxlUimQNSJByIRKm0IwQ?both) for proposals, and submit changes to our branch: https://github.com/DecentralizedScience/blips/tree/blip3

kiview commented on blips/blip-3-peerreview.md:
```diff
@@ -0,0 +1,53 @@
+|     manuscriptHash | string     |     |  Yes    |
+|     recommendation | enum     |     |  Yes    |
+|     url | string     |     |  Yes    | The url should give public access to the content of the review. Alternativelly, it can be a IPFS address so the content is served from a decentralized network
+|     verified | bool     |     |  Yes    |
```
When does the review become verified and who may set this value? Or can this also be a transient value, e.g.
```
isVerified() {
  return vouchers.size > n
}
```

atfornes replied:
I guess different applications and users would like to decide their criteria independently.

Currently, only one "vouche" was needed. In a more decentralized fashion, each actor would decide depending on which identities vouched for the review (stored at vouchersMap)

For instance, our app could only trust reviews verified by us, while other software could require a minimum number, or trust only a collection of actors.

kiview replied:
Makes sense, but what was the meaning of the `verified` field in this case?

atfornes replied:
I would deprecate that field. We added it because current implementation has it.

kiview commented on blips/blip-3-peerreview.md:
```diff
@@ -0,0 +1,53 @@
+- [ ] **TODO** Write a collection of existing standards: Crossref, Orcid, Publons, https://casrai.org/ (ORCid uses casrai, they are preparing a ISO standard on that standard proposal), etc.
+
+
+## Decentralized Vouchers for Review Validity
```
As mentioned in Slack, I would prefer to replace `Voucher` with `Verifier`.  Another idea is `Endorser`, this is the term LinkedIn uses for endorsing skills of your contacts.

atfornes replied:
I **strongly** agree

Commit:
Several changes

Adding some changes from the discussion at Slack group and github pubb request.
- Changing "Voucher" to "Endorser",
- Adding incomplete functions table
- adding formats for several fields
- and more

Co-authored-by: Kevin Wittek <kiview@users.noreply.github.com>
Co-authored-by: Sebastian Müeller <semuelle@users.noreply.github.com>
Co-authored-by: James Lawton <crossoveranx@users.noreply.github.com>

JamesLawton commented on blips/blip-3-peerreview.md:
```diff
@@ -27,27 +29,37 @@ This standard should specify which fields to register in every record, the forma
+|     publisher | string     |     Name?   |  Yes    |
+|     manuscriptId | string     |  DOI   |  Yes    |
+|     manuscriptHash | string     |  IPFS   |  Yes    |
+|     recommendation | enum (**TODO**: See other standards such as Orcid, Publons...)    |     |  Yes    |
```
May be interesting to have an additional field that labels the type of peer review it is, i.e., double-blind, open, etc.

JamesLawton commented on blips/blip-3-peerreview.md:
```diff
@@ -27,27 +29,37 @@ This standard should specify which fields to register in every record, the forma
-### How to bring trust to public vouchers
-As a decentralized application where we would allow different vouchers, users and applications should decide which vouchers to trust. A system where users could vote up or down specific voucher's addresses could help to build trust in such addresses. Also, we could use the more advanced Token Curated Registries.
+### How to bring trust to public endorsers
+As a decentralized application where we would allow different endorsers. Users, applications and other actors should decide which endorsers to trust. A system where users could vote up or down specific endorser's addresses could help to build trust in such addresses. Also, we could use the more advanced Token Curated Registries as a decentralized reputation system for endorses identities.
```
If create a more robust reputation system for establishing trust in endorses, we should describe it thoroughly in the BLIP. This will likely be a critical aspect of a decentralized peer review process.

atfornes replied:
We can keep it simple for the first BLIP proposal, and just have a list of endorsers. Several reputation systems for endorsers can later be defined, maybe as other BLIPs.

JamesLawton committed:
updated purpose and guidelines and research object cert

JamesLawton committed:
updated purpose and guidelines blip and readme

Commit:
add Blip 3 to index

Commit:
Merge branch 'master' into blip3

kuzdogan commented on blips/blip-3-peerreview.md:
```diff
@@ -0,0 +1,65 @@
+
+- Orcid: https://support.orcid.org/hc/en-us/articles/360006971333-Peer-Review
+- Crossref: https://www.crossref.org/education/content-registration/recommendations-by-content-type/peer-reviews/
+
```
Maybe add https://data.crossref.org/reports/help/schema_doc/4.4.2/schema_4_4_2.html#peer_review

atfornes replied:
sure, please add it to the document!
